web/third/bootstrap-typeahead/typeahead.js

```diff
@@ -503,9 +503,12 @@ import {get_string_diff} from "../../src/util";
     }
 
   , keypress: function (e) {
-      if (this.suppressKeyPressRepeat) return
-      this.move(e)
+    if (!this.suppressKeyPressRepeat) {
+      this.move(e);
+      return;
     }
+    this.maybeStopAdvance(e);
+  }
```

Comment:
Can you explain this commit more? I also don't understand how to quickly check whether it affects other typeaheads or just this code path.

Reply:
The change is that `if (this.suppressKeyPressRepeat)` we `this.maybeStopAdvance(e);`, when we didn't before.

The code to stop advance was previously present in `keyup` but not `keypress`, and I don't see any reason why that distinction needs to be made. 

This does affect all typeaheads theoretically, but it only has an impact when `advanceKeyCodes` is nonempty, which only seems to show up in `search.js`. So effectively it only affects search right now.

Reply:
OK, that seems safe enough.